torch/distributed/rpc/backend_registry.py

```diff
@@ -21,14 +22,56 @@ def register_backend(backend_name, init_backend_handler):
             `_init_rpc()` function is called with a backend.
              This returns the agent.
     """
-    backend_registry = _get_backend_registry()
-    if backend_name in backend_registry:
+    global BackendType
+    if backend_name in BackendType.__members__.keys():
         raise RuntimeError("RPC backend {}: already registered".format(backend_name))
-    backend_registry[backend_name] = init_backend_handler
+    # Create a new enum type, `BackendType`, with extended members.
+    existing_enum_dict = {member.name: member.value for member in BackendType}
+    extended_enum_dict = dict(
+        {backend_name: BackendValue(init_backend_handler=init_backend_handler)},
+        **existing_enum_dict
+    )
+    BackendType = enum.Enum(value="BackendType", names=extended_enum_dict)
+    return BackendType[backend_name]
+
+
+def init_backend(backend, *args, **kwargs):
+    return backend.value.init_backend_handler(*args, **kwargs)
+
+
+def process_group_init_backend_handler(
+    store,
+    self_name,
+    self_rank,
+    worker_name_to_id,
+    num_send_recv_threads,
+    *args,
+    **kwargs
+):
+    # Initialize ProcessGroup.
+    if not dc10d.is_initialized():
+        world_size = len(worker_name_to_id)
+        dc10d.init_process_group(
```

Comment:
This is a side effect of this method. Shall we add docs to `init_model_parallel` to describe this?

Reply:
I feel a better solution might be letting ProcessGroupAgent to have its own pg instance instead of modifying the global one. But I am OK to land this as is to unblock.

Reply:
Yes, this will caues confusion to users. We could address this later. 

The work comes from the fact that all helper functions in `distributed_c10d.py` modify the global varaiables like `_default_pg` and `_pg_map`. We need to refactor distributed_c10d.py` a bit to have basic building block that only returns a `pg` withought manipulate the global variables. Then, put a `_model_parallel_pg` in `model_parallel/__init__.py`